### ISSUE TITLE
Add highlighting in results

### ISSUE THREAD

**Issue by sylvinus** [commonsearch/cosr-front#22]
Having parts of the results which matched in bold is a great visual aid.

An open question is: should we highlight results from the API as well? Or is that left to the client (in that case, to our JavaScript as well) ?

**Comment by JBaba** [comment]
I think If we choose to do it in JavaScript then it comes to do post processing (after all search results are rendered). 
But if it comes from the api then I think even static links will have that functionalities that's a plus. (am i wrong here ?)
one question when you say API means you are referring to GoLang or Python ?

**Comment by sylvinus** [comment]
Yes, static links should be as close in functionality as possible, so we probably want to do this at the API /  Go level, not JavaScript.

**Comment by JBaba** [comment]
@sylvinus for this I am thinking doing something like this,

```
hitsByIds[hit.Id] = &Hit{
            ID:     ---,
            URL:    ----,   
            Title:   strings.Replace(hitsByIds[hit.Id].Title, " "+req.Query+" ", " <b>"+req.Query+"</b> ",-1),
            Summary: strings.Replace(hitsByIds[hit.Id].Summary, " "+req.Query+" ", " <b>"+req.Query+"</b> ",-1)}
```

This does not replaces map value but however,
If i run independent line it gives me proper replaced text,

`fmt.Println(strings.Replace(hitsByIds[hit.Id].Title, " "+req.Query+" ", " <b>"+req.Query+"</b> ",-1))`

Am i doing something wrong ? or you suggest this approach or something else ?

**Comment by sylvinus** [comment]
This is a good start! You should really have a separate `AddHighlighting()` function that will grow over time as we include more improvements and edge cases!
